jina/executors/decorators.py

```diff
@@ -365,3 +372,95 @@ def arg_wrapper(*args, **kwargs):
         return _batching(func)
     else:
         return _batching
+
+def batching_ranker_input(func: Callable[[Any], np.ndarray] = None,
+                         batch_size: Union[int, Callable] = None,
+                         num_batch: Optional[int] = None,
+                         slice_on: Union[int, List[int]] = 2,
+                         num_data: int = None) -> Any:
```

Comment:
num_data default is 3 I think for this case no?

Reply:
out of `self, query_meta, old_match_scores, match_meta` we only want to batch `old_match_scores` . other two are meta which dont need to batched.